payments/models.py

```diff
@@ -154,3 +169,309 @@ def __str__(self):
             f"{self.service} - {self.period} "
             f"({self.price_value}{'€' if self.price_unit == PriceUnits.AMOUNT else '%'})"
         )
+
+
+class OrderManager(models.Manager):
+    def berth_orders(self):
+        return self.get_queryset().filter(
+            Q(
+                Q(_product_content_type__model=BerthProduct._meta.model_name)
+                | Q(_lease_content_type__model=BerthLease._meta.model_name)
+            )
+        )
+
+    def winter_storage_orders(self):
+        return self.get_queryset().filter(
+            Q(
+                Q(_product_content_type__model=WinterStorageProduct._meta.model_name)
+                | Q(_lease_content_type__model=WinterStorageLease._meta.model_name)
+            )
+        )
+
+
+class Order(UUIDModel, TimeStampedModel):
+    customer = models.ForeignKey(
+        "customers.CustomerProfile",
+        verbose_name=_("customer"),
+        on_delete=models.CASCADE,
+        related_name="orders",
+    )
+    product = GenericForeignKey("_product_content_type", "_product_object_id")
+    lease = GenericForeignKey("_lease_content_type", "_lease_object_id")
+    status = EnumField(OrderStatus, default=OrderStatus.WAITING)
+    comment = models.TextField(blank=True, null=True)
+    base_price = models.DecimalField(
+        verbose_name=_("price"),
+        max_digits=10,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.01"))],
+        blank=True,
+        null=True,
+    )
+    tax_percentage = models.DecimalField(
+        verbose_name=_("tax percentage"),
+        max_digits=5,
+        decimal_places=2,
+        blank=True,
+        null=True,
+    )
+    due_date = models.DateField(
+        verbose_name=_("due date"), default=calculate_order_due_date,
+    )
+
+    _product_content_type = models.ForeignKey(
+        ContentType,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="product",
+    )
+    _product_object_id = models.UUIDField(
+        null=True, blank=True, verbose_name=_("product")
+    )
+
+    _lease_content_type = models.ForeignKey(
+        ContentType,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="lease",
+    )
+    _lease_object_id = models.UUIDField(null=True, blank=True, verbose_name=_("lease"))
+
+    objects = OrderManager()
+
+    def __str__(self):
+        return f"{self.product} [{self.status}]"
+
+    @property
+    @rounded
+    def full_price(self):
+        return convert_pretax_to_aftertax(self.base_price, self.tax_percentage)
```

Comment:
I think this is the other way around: we want to store taxful price and then subtract taxes programmatically 🤔 
e.g. the orders inported from Timmi will have their total taxful price and tax percentage

Reply:
What would be the the advantage of having taxful vs taxless? Is it only the Timmi import? _In my mind_, it makes more sense to store the actual base price, but what say you?

Reply:
as per Slack discussion these are now changed to `price with tax` and `tax percentage`. thread is resolved 👍